packages/collector/src/agentConnection.js

```diff
@@ -264,8 +264,15 @@ exports.sendMetrics = function sendMetrics(data, cb) {
  */
 exports.sendSpans = function sendSpans(spans, cb) {
   const callback = util.atMostOnce('callback for sendSpans', err => {
+    let spanInfo;
     if (err && !maxContentErrorHasBeenLogged && err instanceof PayloadTooLargeError) {
       logLargeSpans(spans);
+    } else if (err && !maxContentErrorHasBeenLogged) {
+      spanInfo = getSpanLengthInfo(spans);
+      logger.warn('error occurred in sendSpans. no of spans found: %s', spanInfo);
+    } else {
+      spanInfo = getSpanLengthInfo(spans);
+      logger.debug('debugging in sendSpans. no of spans found: %s', spanInfo);
```

Comment:
Success case (debug)
> Successfully sent: { exitSpans: 2, entreySpans: 1 }

Error case (warns)
> Failed to send: { exitSpans: 2, entreySpans: 1 } (agentConnection)
> Failed to transmit spans, will retry ... (spanBuffer)
> Failed to send: { exitSpans: 2, entreySpans: 1 } (agentConnection)
> Failed to transmit spans, will retry ... (spanBuffer)

The logs from the error case are warnings. They always appear on the customers application in case there is an error. IMO these two logs are very much the same. We should always try to log as little as possible in "normal" log modus.

Reply:
In that case, should we keep the logs in `debug` mode only? 
The log can be added without checking whether its an error or not

Reply:
Yes good idea. Lets use debug!

Reply:
Ideally we do not even generate the span information in **non** debug modus.